crates/external_services/src/grpc_client.rs

```diff
@@ -128,3 +128,13 @@ impl<T> AddHeaders for tonic::Request<T> {
         });
     }
 }
+
+#[cfg(feature = "dynamic_routing")]
+pub(crate) fn create_grpc_request<T: Debug>(message: T, headers: GrpcHeaders) -> tonic::Request<T> {
+    let mut request = tonic::Request::new(message);
+    request.add_headers_to_grpc_request(headers);
```

Comment:
How do the error logs thrown by `add_headers_to_grpc_request()` look like? Do they have the file and line number information that helps identify where exactly it was called, or should we consider adding `#[track_caller]` to `create_grpc_request()` as well?